src/Features/Core/Shared/Utilities/LinkedFilesSymbolEquivalenceComparer.cs

```diff
@@ -16,7 +17,14 @@ internal sealed class LinkedFilesSymbolEquivalenceComparer : IEqualityComparer<I
 
         bool IEqualityComparer<ISymbol>.Equals(ISymbol x, ISymbol y)
         {
-            return x.Kind == y.Kind && x.Name == y.Name;
+            return KindsAreEquivalent(x.Kind, y.Kind) && x.Name == y.Name;
+        }
+
+        private bool KindsAreEquivalent(SymbolKind x, SymbolKind y)
+        {
+            return x == y  
+                   || (x == SymbolKind.Property && y == SymbolKind.Field)
+                   || (x == SymbolKind.Field && y == SymbolKind.Property);
```

Comment:
I felt uneasy relaxing the check to be name-only, but maybe that's what we really want.